libraries/standard/mqtt/src/mqtt.c

```diff
@@ -1384,58 +1502,57 @@ MQTTStatus_t MQTT_ProcessLoop( MQTTContext_t * const pContext,
 
     while( status == MQTTSuccess )
     {
-        status = MQTT_GetIncomingPacketTypeAndLength( pContext->transportInterface.recv,
-                                                      pContext->transportInterface.networkContext,
-                                                      &incomingPacket );
+        status = receiveSingleIteration( pContext, remainingTimeMs, true );
 
-        if( status == MQTTNoDataAvailable )
-        {
-            /* Assign status so an error can be bubbled up to application,
-             * but reset it on success. */
-            status = handleKeepAlive( pContext );
-
-            if( status == MQTTSuccess )
-            {
-                /* Reset the status to indicate that we should not try to read
-                 * a packet from the transport interface. */
-                status = MQTTNoDataAvailable;
-            }
-        }
-        else if( status != MQTTSuccess )
+        if( status != MQTTSuccess )
         {
-            LogError( ( "Receiving incoming packet length failed. Status=%d",
-                        status ) );
+            LogError( ( "Exiting process loop. Error status=%d", status ) );
```

Comment:
The loop does exit since the status is checked in the loop condition. There shouldn't be any problem with checking the elapsed time since the loop will exit afterwards anyway. We could put a `break` here to be explicit that the loop will exit, but that would be a MISRA violation since we already have one `break` in the loop